src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -153,6 +149,63 @@ public ResourceTypeSchema(Schema schema) {
             });
             return null;
         });
+
+        this.unprocessedProperties.computeIfPresent("tagging", (k, v) -> {
+            ((Map<?, ?>) v).forEach((key, value) -> {
+                if (key.equals("taggable") || key.equals("tagOnCreate") || key.equals("tagUpdatable")
+                    || key.equals("cloudFormationSystemTags")) {
+                    this.tagging.put(key.toString(), Boolean.valueOf(value.toString()));
+                } else if (key.equals("tagProperty")) {
+                    this.tagging.put(key.toString(), new JSONPointer(value.toString()));
+                } else {
+                    throw new ValidationException("Unexpected tagging metadata attribute", "tagging", "#/tagging/" + key);
+                }
+            });
+            validateTaggingMetadata();
+            return null;
+        });
+
+        if (this.unprocessedProperties.containsKey("taggable")) {
+            this.taggable = Boolean.parseBoolean(this.unprocessedProperties.get("taggable").toString());
+            if (!this.tagging.containsKey("taggable")) {
+                this.tagging.put("taggable", this.taggable);
+                if (this.taggable) {
+                    // default value to true when taggable to enforce tagging feature
+                    tagging.put("tagOnCreate", tagging.getOrDefault("tagOnCreate", true));
+                    tagging.put("tagUpdatable", tagging.getOrDefault("tagUpdatable", true));
+                    tagging.put("cloudFormationSystemTags", tagging.getOrDefault("cloudFormationSystemTags", true));
+                }
+            } else {
+                throw new ValidationException("More than one configuration found for taggable value." +
+                    " Please remove the deprecated taggable property.", "tagging", "#/tagging/taggable");
+            }
+        } else {
+            this.taggable = true;
+        }
+        this.unprocessedProperties.remove("taggable");
+    }
+
+    private void validateTaggingMetadata() {
```

Comment:
If we have a new class, the validation could be part of that class.

Reply:
yeah already moved it into the ResourceTagging class